modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/test/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchConnectionManagerTest.java

```diff
@@ -271,6 +273,54 @@ public void testAddConnectionWithConnectionIdAndIsNotActive() {
 		).connect();
 	}
 
+	@Test
+	public void testApplyProxyConfigurationInElasticsearchConnection() {
+		_enableRemoteMode();
+
+		String[] networkHostAddresses = RandomTestUtil.randomStrings(10);
+		String proxyHost = RandomTestUtil.randomString();
+		String proxyPassword = RandomTestUtil.randomString();
+		int proxyPort = RandomTestUtil.randomInt();
+		String proxyUserName = RandomTestUtil.randomString();
+
+		Mockito.when(
+			_elasticsearchConfigurationWrapper.networkHostAddresses()
+		).thenReturn(
+			networkHostAddresses
+		);
+		Mockito.when(
+			_elasticsearchConfigurationWrapper.proxyHost()
+		).thenReturn(
+			proxyHost
+		);
+		Mockito.when(
+			_elasticsearchConfigurationWrapper.proxyPassword()
+		).thenReturn(
+			proxyPassword
+		);
+		Mockito.when(
+			_elasticsearchConfigurationWrapper.proxyPort()
+		).thenReturn(
+			proxyPort
+		);
+		Mockito.when(
+			_elasticsearchConfigurationWrapper.proxyUserName()
+		).thenReturn(
+			proxyUserName
+		);
+
+		_elasticsearchConnectionManager.applyConfigurations();
+
+		ProxyConfig proxyConfig = ReflectionTestUtil.getFieldValue(
+			_elasticsearchConnectionManager.getElasticsearchConnection(),
+			"_proxyConfig");
+
+		Assert.assertEquals(proxyHost, proxyConfig.getHost());
+		Assert.assertEquals(proxyPassword, proxyConfig.getPassword());
+		Assert.assertEquals(proxyPort, proxyConfig.getPort());
+		Assert.assertEquals(proxyUserName, proxyConfig.getUserName());
```

Comment:
What about `networkHostAddresses()`?

Reply:
it's just to avoid a NPE. There is no field in ProxyConfig with that name.